### ISSUE TITLE
[1.19.2] Chipped together with ModernFix + Any Structure mod with lots of structures causes worlds to be stuck generating for a long time.

### ISSUE THREAD

**Issue by Gbergz** [embeddedt/ModernFix#73]
**Info:**
When the mod Chipped is paired together with ModernFix + Any Structure mod that has lots of structures. 
It causes worlds to be stuck generating for a long time. Almost forever stuck at 0%.
(In my instance I'm using Rex's Additional Structures to test, since it has lots of structures.)

Upon removing the mod **Chipped**, the issue is no longer happening and worlds generate very fast and normal speed.
So I'm guessing there's something with that mod (Chipped) that ModernFix doesn't like very much.

**Mod List**
**Connected Textures mod**: v1.1.6+8
**Resourceful Lib**: v1.1.24
**Rex's Additional Structures**: v.4.0.3
**ModernFix**: v3.4.1
**Chipped**: v2.1.4

**MC**: 1.19.2
**Forge**: 43.2.8

**-** Gbergz.

**Comment by embeddedt** [comment]
Thanks, fixed in latest beta.

**Comment by Gbergz** [comment]
Thanks! Any idea when that will be live on CurseForge? :)

**Comment by embeddedt** [comment]
Hoping to push a release in the next few days if there are no more bugs found. I just finished a massive refactor to create the Fabric version so more testing is needed than usual.

**Comment by Gbergz** [comment]
Gotcha, thanks for the update either way !

**Comment by embeddedt** [comment]
Fixed in latest releases for all MC versions.

**Comment by Gbergz** [comment]
Thank you, again!

**Comment by Gbergz** [comment]
Update: The latest update to ModernFix made an improvement, but.. it's still really really slow. 
Without Chipped it's fine though like originally reported.

**Comment by embeddedt** [comment]
I can't reproduce that. I tested generating the same world, with the same seed, and it took 92 seconds with ModernFix and 100 seconds without. These were the mods I used:

![image](https://user-images.githubusercontent.com/42941056/236543000-928e0083-d5cf-46c8-89de-19a1c8836efe.png)

It's worth noting that the main reason for the lag at the start of worldgen is that Rex's Additional Structures ships outdated structure files which the game needs to update first (which is extremely laggy). If you first run `/modernfix upgradeStructures` while logged into a singleplayer world, it will generate cached versions of any outdated structures, which should make generating new worlds afterwards much faster.
